tests/suite/cli_misc.rs

```diff
@@ -946,7 +946,10 @@ fn override_by_toolchain_on_the_command_line() {
         );
         config.expect_err(
             &["rustup", "@stable", "which", "rustc"],
-            r#"Invalid value "@stable" for '<+toolchain>': Toolchain overrides must begin with '+'"#,
+            "Invalid value \"@stable\" for '<+toolchain>': \"@stable\" is not a valid subcommand, \
+            so it was interpreted as a toolchain name, but it is also invalid. \
+            To override the toolchain using the 'rustup +toolchain' syntax, \
+            make sure to prefix the toolchain override with a '+'",
```

Comment:
We can probably delete this bit of the integration test; your new test confirms the same behaviour